phoenix-core-client/src/main/java/org/apache/phoenix/schema/PMetaDataImpl.java

```diff
@@ -162,7 +162,11 @@ public void addTable(PTable table, long resolvedTime) throws SQLException {
             metaData.put(table.getKey(), tableRef);
         }
         for (PTable index : table.getIndexes()) {
-            metaData.put(index.getKey(), tableRefFactory.makePTableRef(index, this.timeKeeper.getCurrentTime(), resolvedTime));
+            PTable indexPTable = index;
```

Comment:
We add an index to the cache at L160 and L162 above. 
There is a specific check whether the table is an index or not at L133 above. 
If the table  is an index, then codepath between L133-L156 is executed where we will miss adding the parent link to the cache.